buildkite/run_check_deploy_docker_build.sh

```diff
@@ -4,20 +4,10 @@
 # built successfully. This check is done in the buildkite-assets repo to prevent
 # an unbuildable Dockerfile from being deployed to buildkite instances.
 
-DOCKER_IMAGE_NAME=check-heroku-deploy-image
+echo "--- Building and testing with panorama_command.sh"
+HEROKU_DEPLOY_PATH=deploy/heroku-deploy \
+  HEROKU_DEPLOY_IMAGE_NAME=check-heroku-deploy-image \
+  DEPLOYMENT_APP_NAME=panorama-addons \
+  bash deploy/panorama_command.sh custom
 
-echo "--- Running heroku-deploy build"
-docker build \
-  --build-arg buildkite_agent_uid=$UID \
-  -t $DOCKER_IMAGE_NAME /usr/local/bin/heroku-deploy > build_results.out
```

Comment:
Previously, this build step would build the Docker image we use on build agents, but this `docker build` command refers to the `Dockerfile` in `/usr/local/bin/heroku-deploy`. We only copy data from this repo to that directory as [part of `bootstrap.sh`](https://github.com/panorama-ed/buildkite-assets/pull/199/files#diff-99d5a2a8ca4290b61836b3ad37941f2b925e9ff3151c90dcd5ea82364e4122a1L21), which only gets run when a build agent first starts; that meant that this step was actually only testing the `Dockerfile` that was in the `main` branch at the time the agent running the step booted, and not the `Dockerfile` that's part of the actual branch being built by Buildkite. See [this build](https://buildkite.com/panorama-education/buildkite-assets/builds/293#0184d00e-0b7a-4470-98a0-5480cef8a2aa/135-137) for confirmation; it's using `heroku/heroku:18` as the base image, but [the `Dockerfile` in the commit that build was for](https://github.com/panorama-ed/buildkite-assets/blob/b9c775b8e6973720186fde6565d631255af5767f/deploy/heroku-deploy/Dockerfile#L1) uses `ruby:3.1.3` as the base image. This step just re-built the old Docker container instead of the new one.

To fix this so we're actually testing the new `Dockerfile`, I first changed this from `/usr/local/bin/heroku-deploy` to `deploy/heroku-deploy`, which is the relative path within the checked-out build commit where the new `Dockerfile` lives. That worked, but once I was in here I decided that it would be nice to test not only that the docker container builds successfully, but also that we can run `panorama_command.sh` with the new container since that's what our Buildkite setup actually uses in general. Looking in that script, we see that internally [it already builds the container](https://github.com/panorama-ed/buildkite-assets/blob/main/deploy/panorama_command.sh#L23) and supports specifying a `Dockerfile` location via the `HEROKU_DEPLOY_PATH` env var, so I changed this script to just run `panorama_command.sh` against the build commit's updated `Dockerfile`. This covers the existing functionality as well as testing that `panorama_command.sh` workers with the new container, so it seems like a solid improvement.

Reply:
That makes sense. Thanks for writing this up. I agree all we need to do is verify the container script works with `panorama_command.sh custom`. We wouldn't be able to test out `panorama_command.sh` with `migrate` and `deploy` without app code.
If we pushed bad code on [line 37 here](https://github.com/panorama-ed/buildkite-assets/blob/main/deploy/panorama_command.sh#L37), e.g. setting `command` to error out, we should see that the next time other apps are deployed. That error should _hopefully_ indicate to other engineers that `buildkite-assets` is the issue.